src/git.ts

```diff
@@ -62,15 +64,22 @@ async function gitRemoteNameAndBranch(repoDirectory: string): Promise<RemoteName
     let remoteName: string | undefined
     let branch = 'HEAD'
 
-    const { stdout } = await execa('git', ['rev-parse', '--abbrev-ref', 'HEAD@{upstream}'], { cwd: repoDirectory })
-    const remoteAndBranch = stdout.split('/')
+    try {
+        const { stdout } = await execa('git', ['rev-parse', '--abbrev-ref', 'HEAD@{upstream}'], { cwd: repoDirectory })
+        const remoteAndBranch = stdout.split('/')
 
-    if (remoteAndBranch.length === 1) {
-        // The upstream branch points to a local branch.
-        ;[remoteName] = remoteAndBranch
-    }
-    if (remoteAndBranch.length === 2) {
-        ;[remoteName, branch] = remoteAndBranch
+        // TODO(tj): can't the remote have a slash as well?
+
+        if (remoteAndBranch.length === 1) {
+            // The upstream branch points to a local branch.
+            ;[remoteName] = remoteAndBranch
+        }
+        if (remoteAndBranch.length >= 2) {
+            ;[remoteName] = remoteAndBranch
+            branch = stdout.slice(remoteName.length + 1) // remove '/' from start of branch
+        }
+    } catch {
+        // noop. upstream may not be set
```

Comment:
Can you narrow down this error a bit so we don't blanket-swallow all errors, including e.g. the `git` executable not being found? Is there a certain exit code or stderr output?

Reply:
Git doesnt seem to return [specific error codes](https://stackoverflow.com/questions/4917871/does-git-return-specific-return-error-codes/4918002#4918002).
Some examples:
<img width="599" alt="Screen Shot 2021-02-16 at 11 09 16 AM" src="https://user-images.githubusercontent.com/37420160/108090812-0ee8e680-7049-11eb-8b11-ddcbc89fccb6.png">
<img width="489" alt="Screen Shot 2021-02-16 at 11 07 34 AM" src="https://user-images.githubusercontent.com/37420160/108090822-127c6d80-7049-11eb-9726-abe892c771c2.png">


If stderr starts with `fatal: no upstream configured for branch` we could swallow, else rethrow. 

However, errors like `git` executable not being found or git repo not existing would be thrown when we execute `git.remotes` or `git.branches` (both are executed before `git.upstreamAndBranch`)